test/es-module/test-esm-import-flag.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+require('../common');
+const fixtures = require('../common/fixtures');
+const assert = require('assert');
+const { spawnSync } = require('child_process');
+
+{
+  // known limitation:
+  // imports should not be imported when evaluating text
```

Comment:
Regarding `--print`: because of TLA I assume, same as when using `--input-type=module`. Regarding `--eval`, I agree that feels like a bug.

Reply:
If there’s a good reason why it shouldn’t be supported, we should just error whenever the two incompatible flags are used together. Otherwise I think we should try to add support (for `--eval`, say) within this PR unless it’s such a substantial effort that it justifies a separate PR.

Reply:
this is not a bug, and it will work when adding `--input-type=module`

Reply:
do you thing we should change?
https://github.com/nodejs/node/blob/c7222b35899ad352d78f288ddc16c6d19ca79a06/lib/internal/main/eval_string.js#L24-L25
to be
```js
if (getOptionValue('--input-type') === 'module' || getOptionValue('--import').length)
  evalModule(source, print);
```

Reply:
Can't it simply eval the string as CJS? I don't think it should have any effect on how we eval a string (or am I missing something?).

Reply:
possibly, I will look into that

Reply:
converting `eval_string` to having an async bootstrap flow will indeed break some stuff:
originally the behavior was
```js
node  -e "let ______________________________________________; throw 10"      

[eval]:1
let ______________________________________________; throw 10
                                                    ^
10
```
after the changes, it becomes
```js
./node  -e "let ______________________________________________; throw 10"

node:internal/process/esm_loader:110
    internalBinding('errors').triggerUncaughtException(
                              ^
10
```

wich is the same behavior of running with `--input-type="module"` before my changes:

```js
node --input-type="module" -e "let ______________________________________________; throw 10"

node:internal/process/esm_loader:94
    internalBinding('errors').triggerUncaughtException(
                              ^
10
```

so I am wondering if this is ok to break or not, and if not - what can be done to avoid that? since it was previously a synchronous bootstrap flow

Reply:
IMHO nothing should change when not using the `--import` flag, the changes should only apply when using the flag.

Reply:
done